ibmcloud: support e2e test overlay yaml

huoqifeng opened the pull request:
Fixes: #712

liudalibj reviewed (approved):
LGTM

stevenhorsman commented on test/provisioner/provision_ibmcloud_initializer.go:
```diff
@@ -81,14 +81,13 @@ func initLogger() {
+		CosServiceURL:   properties["COS_SERVICE_URL"],
+		IamServiceURL:   properties["IAM_SERVICE_URL"],
 		InstanceProfile: properties["INSTANCE_PROFILE_NAME"],
 		PodvmImageID:    properties["PODVM_IMAGE_ID"],
```
I think that `PODVM_IMAGE_ID` has been removed from the default `provision_ibmcloud.properties` files, so maybe we need to add it back. I appreciate that longer term we might want to string `./cluster-provisioner -action=uploadimage` into provision and be able to re-use the output, but I don't think we are quite ready for that yet?

stevenhorsman commented:
@huoqifeng - I think this might need rebased on latest staging as I hit the error `IAM_SERVICE_URL was not set, example: https://iam.cloud.ibm.com/identity/token` when testing it, which I think you removed/fixed earlier this week?

stevenhorsman commented:
After pulling this properly I've hit an issue:
```
 ./cluster-provisioner -action=provision
INFO[0000] IamServiceURL is: https://iam.cloud.ibm.com/identity/token.
INFO[0000] VpcServiceURL is: https://jp-tok.iaas.cloud.ibm.com/v1.
Validation Failure for field: name
```
I'm guessing it is user error and I've not supplied a properties properly, so I'll debug it later, but adding some extra info might be helpful :)

huoqifeng commented:
I met some issues when run the e2e test, mark it as `WIP`. Thanks @stevenhorsman and @liudalibj review, I need some more time to fix the issues found.

huoqifeng commented:
Tested as below:
```
# export LOG_LEVEL="debug"
# export CLOUD_PROVIDER="ibmcloud"
# export TEST_E2E_PROVISION_FILE="/root/provision_ibmcloud.properties"
# export TEST_E2E_PROVISION="yes"
# export TEST_E2E_TIMEOUT=120m
# make test-e2e
go test -v -tags=ibmcloud -timeout 120m -count=1 ./test/e2e
time="2023-03-16T21:16:46-07:00" level=debug msg="&{ApiKey: ...}"
time="2023-03-16T21:16:46-07:00" level=info msg="IamServiceURL is: https://iam.cloud.ibm.com/identity/token."
time="2023-03-16T21:16:46-07:00" level=info msg="VpcServiceURL is: https://jp-tok.iaas.cloud.ibm.com/v1."
Do setup
Cluster provisioning
time="2023-03-16T21:16:48-07:00" level=info msg="SSH Key e2e-test-sshkey-123 with ID r022-3eef55dc-2dd5-4ae6-9c7c-c966d3b1846d exists already, we can just use it."
time="2023-03-16T21:16:49-07:00" level=info msg="VPC e2e-test2-vpc with ID r022-4035acde-4b4f-486b-931a-6ac5ff211f00 exists alread"
time="2023-03-16T21:16:50-07:00" level=info msg="Got VPC default SecurityGroupID r022-16964fab-1c6e-458d-a9e9-cc1bbcd1a179."
time="2023-03-16T21:16:51-07:00" level=info msg="Subnet e2e-test2-vpc-subnet with ID 02e7-fe13bcdb-28be-476b-b615-2bdfacc86b33 exists already."
time="2023-03-16T21:16:51-07:00" level=info msg="PublicGateway r022-08ddd4e3-ab7c-4279-a8b5-51b02d0a46e2 exists in subnet."
time="2023-03-16T21:16:52-07:00" level=info msg="Cluster e2e-test2. exists already."
time="2023-03-16T21:16:52-07:00" level=info msg="Waiting for cluster e2e-test2 to be available."
time="2023-03-16T21:16:53-07:00" level=info msg="Cluster e2e-test2 is available."
time="2023-03-16T21:16:53-07:00" level=info msg="Sync cluster kubeconfig to /root/.kube with current config context"
time="2023-03-16T21:16:56-07:00" level=info msg="Downloaded cluster e2e-test2 kube configuration into folder /root/.kube"
time="2023-03-16T21:16:56-07:00" level=debug msg="{AdminKey:...}"
Deploy the Cloud API Adaptor
Install the controller manager
Wait for the cc-operator-controller-manager deployment be available
Customize the overlay yaml file
time="2023-03-16T21:17:06-07:00" level=debug msg="map[CLOUD_PROVIDER:ibmcloud ...]"
Install CoCo and cloud-api-adaptor
Wait for the cc-operator-daemon-install DaemonSet be available
Wait for the pod cc-operator-daemon-install-zng5m be ready
Wait for the cloud-api-adaptor-daemonset DaemonSet be available
Wait for the pod cloud-api-adaptor-daemonset-mvpkl be ready
Wait for the kata runtimeclass be created
=== RUN   TestCreateSimplePod
=== RUN   TestCreateSimplePod/Simple_Peer_Pod
    common_suite_test.go:35: timed out waiting for the condition
--- FAIL: TestCreateSimplePod (180.62s)
    --- FAIL: TestCreateSimplePod/Simple_Peer_Pod (180.62s)
FAIL
time="2023-03-16T21:20:35-07:00" level=info msg="Deleting Cluster e2e-test2."
time="2023-03-16T21:20:38-07:00" level=info msg="Waiting for cluster e2e-test2 to be removed..."
time="2023-03-16T21:20:38-07:00" level=info msg="Waited 0 minutes..."
time="2023-03-16T21:21:39-07:00" level=info msg="Waited 1 minutes..."
time="2023-03-16T21:22:39-07:00" level=info msg="Waited 2 minutes..."
time="2023-03-16T21:23:40-07:00" level=info msg="Waited 3 minutes..."
time="2023-03-16T21:24:41-07:00" level=info msg="Waited 4 minutes..."
time="2023-03-16T21:25:41-07:00" level=info msg="Waited 5 minutes..."
time="2023-03-16T21:26:42-07:00" level=info msg="Waited 6 minutes..."
time="2023-03-16T21:27:42-07:00" level=info msg="Cluster e2e-test2 is removed."
time="2023-03-16T21:27:44-07:00" level=info msg="SSH Key e2e-test-sshkey-123 with ID r022-3eef55dc-2dd5-4ae6-9c7c-c966d3b1846d is deleted."
time="2023-03-16T21:27:45-07:00" level=info msg="Found subnet e2e-test2-vpc-subnet with ID 02e7-fe13bcdb-28be-476b-b615-2bdfacc86b33."
time="2023-03-16T21:27:47-07:00" level=info msg="Public Gatwway r022-08ddd4e3-ab7c-4279-a8b5-51b02d0a46e2 was unattached from Subnet 02e7-fe13bcdb-28be-476b-b615-2bdfacc86b33."
time="2023-03-16T21:27:48-07:00" level=info msg="Public Gatwway r022-08ddd4e3-ab7c-4279-a8b5-51b02d0a46e2 was deleted."
time="2023-03-16T21:27:48-07:00" level=info msg="Deleting subnet with ID 02e7-fe13bcdb-28be-476b-b615-2bdfacc86b33."
time="2023-03-16T21:27:49-07:00" level=info msg="Deleted subnet with ID 02e7-fe13bcdb-28be-476b-b615-2bdfacc86b33."
time="2023-03-16T21:27:51-07:00" level=info msg="Found VPC e2e-test2-vpc with ID r022-4035acde-4b4f-486b-931a-6ac5ff211f00."
time="2023-03-16T21:27:51-07:00" level=info msg="Deleting VPC with ID r022-4035acde-4b4f-486b-931a-6ac5ff211f00."
time="2023-03-16T21:27:52-07:00" level=info msg="Deleted VPC with ID r022-4035acde-4b4f-486b-931a-6ac5ff211f00."
```

huoqifeng commented:
For reviewers, there are several changes on this PR:
- Refined cluster-provisioner to support operator install
- Enhanced KustomizeOverlay to support SecretGeneratorLiteral
- Fixed some bug in provision.go
- Added properties `KUBE_VERSION` and `WORKER_OPERATION_SYSTEM` so that we can support s390x
- Added public gateway in subnet for external images pulling
- Overlay yaml support and install operator
- Set DisablePublicServiceEndpoint to false so that we can run kubectl clients

huoqifeng commented:
Still there is one TODO in future:
The cluster-provisioner can only be run in folder test/e2e as the yamls used in provisioner is relative path, we can refine it in future PR.

huoqifeng commented:
@liudalibj @stevenhorsman could you help review it?

stevenhorsman commented on cmd/cluster-provisioner/main.go:
```diff
@@ -32,37 +37,62 @@ func main() {
 
 	if *action == "provision" {
-		if err := provisioner.CreateVPC(context.TODO(), nil); err != nil {
+		fmt.Println("Creating VPC...")
```
I wonder if the `Println` here would be better as log-rs logs instead? I appreciate that it's fitting in with some of the existing logging elsewhere, but maybe we should adapt that? @wainersm - do you have any thoughts on this?

huoqifeng replied:
I agree @stevenhorsman I'd like to use a new PR to use logrus in the e2e test main package and the cluster-provisioner main package. Maybe even other CAA packages via yet another PR.

stevenhorsman commented:
> After pulling this properly I've hit an issue:
> 
> ```
>  ./cluster-provisioner -action=provision
> INFO[0000] IamServiceURL is: https://iam.cloud.ibm.com/identity/token.
> INFO[0000] VpcServiceURL is: https://jp-tok.iaas.cloud.ibm.com/v1.
> Validation Failure for field: name
> ```
> 
> I'm guessing it is user error and I've not supplied a properties properly, so I'll debug it later, but adding some extra info might be helpful :)

After quite a lot of debugging and adding trace I've found the problem here.
In my properties file I specified an ssh_key_id, but not the name. The createSshKey logic (using findSshKey) only looks for name matches and compares them to the id, so it wasn't finding my existing key. I'm guessing this is something we could/should add support for to make it easier to use a pre-existing key? Not a blocker for this PR though as once I added the matching ssh key name provisioning worked.

It also highlighted to me that extra trace would be useful for servicability, but that spans way beyond just this PR!

stevenhorsman commented:
QiFeng, I hit a failure in the test the same as you did above:
```
Deploy the Cloud API Adaptor
Install the controller manager
Wait for the cc-operator-controller-manager deployment be available
Customize the overlay yaml file
time="2023-03-17T05:41:25-07:00" level=debug msg="map[CLOUD_PROVIDER:ibmcloud CRI_RUNTIME_ENDPOINT:/run/cri-runtime/containerd.sock IBMCLOUD_API_KEY:vFFzRY3W_3xhp0FHLmeyKnunNeMt7Mc61Jx-yFH5sCeD IBMCLOUD_IAM_ENDPOINT:https://iam.cloud.ibm.com/identity/token IBMCLOUD_PODVM_IMAGE_ID:r022-7c0f2181-a8b5-45e2-b02f-daef85c28534 IBMCLOUD_PODVM_INSTANCE_PROFILE_NAME:bz2-2x8 IBMCLOUD_RESOURCE_GROUP_ID:31af14bcb3dc4a6499c90b18efb50ea6 IBMCLOUD_SSH_KEY_ID:r022-b779f94d-b8aa-4b91-892c-1515bbc775fc IBMCLOUD_VPC_ENDPOINT:https://jp-tok.iaas.cloud.ibm.com/v1 IBMCLOUD_VPC_ID:r022-a3014538-cd8e-4042-84fc-53f903c84fe1 IBMCLOUD_VPC_SG_ID:r022-2a62bfaa-bef3-44ed-b581-351a7cfbf823 IBMCLOUD_VPC_SUBNET_ID:02f7-17eb70f0-a8e6-4dc9-9f49-d743a2c48424 IBMCLOUD_ZONE:jp-tok-2]"
Install CoCo and cloud-api-adaptor
Wait for the cc-operator-daemon-install DaemonSet be available
Wait for the pod cc-operator-daemon-install-klv6w be ready
Wait for the cloud-api-adaptor-daemonset DaemonSet be available
Wait for the pod cloud-api-adaptor-daemonset-c6ffp be ready
Wait for the kata runtimeclass be created
=== RUN   TestCreateSimplePod
=== RUN   TestCreateSimplePod/Simple_Peer_Pod
    common_suite_test.go:35: timed out waiting for the condition
--- FAIL: TestCreateSimplePod (180.86s)
    --- FAIL: TestCreateSimplePod/Simple_Peer_Pod (180.86s)
```

I'll try and debug it now and see if I can work out what went wrong...

yoheiueda commented:
When I reran the test, cluster creation was skipped, and kata runtimeclass was successfully created. Then, I encountered another error.

```
$ make test-e2e
go test -v -tags=ibmcloud -timeout 120m -count=1 ./test/e2e
︙
time="2023-03-17T15:46:48Z" level=info msg="Cluster e2e-test1. exists already."
time="2023-03-17T15:46:48Z" level=info msg="Waiting for cluster e2e-test1 to be available."
time="2023-03-17T15:46:48Z" level=info msg="Cluster e2e-test1 is available."
time="2023-03-17T15:46:48Z" level=info msg="Sync cluster kubeconfig to /root/.kube
︙
Install CoCo and cloud-api-adaptor
Wait for the cc-operator-daemon-install DaemonSet be available
Wait for the pod cc-operator-daemon-install-5l5sm be ready
Wait for the cloud-api-adaptor-daemonset DaemonSet be available
Wait for the pod cloud-api-adaptor-daemonset-kpfhz be ready
Wait for the kata runtimeclass be created
=== RUN   TestCreateSimplePod
=== RUN   TestCreateSimplePod/Simple_Peer_Pod
    common_suite_test.go:35: timed out waiting for the condition
--- FAIL: TestCreateSimplePod (180.10s)
    --- FAIL: TestCreateSimplePod/Simple_Peer_Pod (180.10s)
FAIL
time="2023-03-17T15:50:23Z" level=info msg="Deleting Cluster e2e-test1."
︙
```

stevenhorsman commented:
> When I reran the test, cluster creation was skipped, and kata runtimeclass was successfully created. Then, I encountered another error.
> 
Yes - that is the same error that QiFeng and I are both seeing (as mentioned above). I've re-created my podvm to be built on the latest code (it was about a week old) and re-trying to see if that was the issue. I'll update my comment above with anything new I learn.

stevenhorsman reviewed (approved):
As documented I hit a few issues, which we can break out as separate items, but I managed to get this working:
```
root@stevenh-qfh-test1:~/go/src/github.com/huoqifeng/cloud-api-adaptor# make test-e2e
go test -v -tags=ibmcloud -timeout 120m -count=1 ./test/e2e
time="2023-03-17T09:35:06-07:00" level=debug msg="&{ApiKey:vFFzRY3W_3xhp0FHLmeyKnunNeMt7Mc61Jx-yFH5sCeD Bucket:sh-pp-z-cos-bucket-ca-tor-2-2-23 ClusterName:sh-e2e-test CosApiKey:vFFzRY3W_3xhp0FHLmeyKnunNeMt7Mc61Jx-yFH5sCeD CosInstanceID:crn:v1:bluemix:public:cloud-object-storage:global:a/c5672f2b9bd2fd6cbc81f37056b2e542:e34e4af3-7af9-45d7-98c4-7f63dcde8f6b:: CosServiceURL:s3.ca-tor.cloud-object-storage.appdomain.cloud SecurityGroupID: IamServiceURL: InstanceProfile:bz2-2x8 KubeVersion:1.26.1 PodvmImageID:r022-9fd4ea70-62c2-4b53-978f-3eed725fbd5d PodvmImageArch:s390x PublicGatewayID: Region:jp-tok ResourceGroupID:31af14bcb3dc4a6499c90b18efb50ea6 SshKeyContent:ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDaq78ahrA9BC9rJv7v7rSsr+z7fN8zzbQXnwctZh3lh5+A2MhEjRTW7QumZBxBKaXWfkSBfV9EIU+QVz8dDvWvK7X3P48td+UpbBgAT91HDTswWma+IiKzzDQGcEZ2LUlfNvhESbUg5DQsgzmmn2LbJqoZVf0VzFTqXlgIM78byc8HBdDQJad5VTV2PF06qmCQsA67HnNN5WfentgxNUsOFjBQE245vLhoSihQsA1zU6t82wBxKrbJp8QZsrmXNsKwm3DokjlBlYiy9ve5BMj0mqGdnGuPxWZvah0VtGpZ20sR7VlXmUJOViFCWbgOqVbDVbqEC0qpz0PyLpOv54Zil1v9PL4lUW4MOL8fOpSSr6QakOqVlkaiOdj48TfzRsJJQHVBQoCpMmgjjd25FNK2zki1igtiMzS73jHxPm0h2NwP5lHpTI9sksoNJLkatxcH6Mzi4hYCZamyr4TYXQWNud1QbY9KvFOmDRuC6Em1PzxawsdrIjWQkjbQok8sW80= root@stevenh-qfh-test1.fyre.ibm.com SshKeyID: SshKeyName:qfh-e2e-testkey SubnetName:sh-e2e-test-vpc-subnet SubnetID: VpcName:sh-e2e-test-vpc VpcID: VpcServiceURL: WorkerFlavor:bx2.2x8 WorkerOS:UBUNTU_20_64 Zone:jp-tok-2 WorkerCount:1 IsSelfManaged:false VPC:<nil> ClusterAPI:<nil>}"
time="2023-03-17T09:35:06-07:00" level=info msg="IamServiceURL is: https://iam.cloud.ibm.com/identity/token."
time="2023-03-17T09:35:06-07:00" level=info msg="VpcServiceURL is: https://jp-tok.iaas.cloud.ibm.com/v1."
time="2023-03-17T09:35:06-07:00" level=trace msg="initClustersAPI()"
time="2023-03-17T09:35:06-07:00" level=trace msg="initVpcV1()"
Do setup
Cluster provisioning
time="2023-03-17T09:35:06-07:00" level=trace msg="CreateVPC()"
time="2023-03-17T09:35:06-07:00" level=trace msg="createSshKey()"
time="2023-03-17T09:35:06-07:00" level=trace msg="findSshKey(qfh-e2e-testkey)"
time="2023-03-17T09:35:09-07:00" level=trace msg="Checking SSH Key stevenh-ssh-key-2023-02-07."
time="2023-03-17T09:35:09-07:00" level=trace msg="Checking SSH Key operator-key."
time="2023-03-17T09:35:09-07:00" level=trace msg="here 1"
time="2023-03-17T09:35:09-07:00" level=trace msg="here 2"
time="2023-03-17T09:35:11-07:00" level=info msg="SSH Key qfh-e2e-testkey with ID r022-765cd597-9813-4dd0-8afa-1e846ddf0d08 is created."
time="2023-03-17T09:35:12-07:00" level=info msg="Creating VPC sh-e2e-test-vpc in ResourceGroupID 31af14bcb3dc4a6499c90b18efb50ea6."
time="2023-03-17T09:35:13-07:00" level=info msg="Created VPC with ID r022-ef0d6365-b867-4c18-adaa-61fc17a71cd2 in ResourceGroupID 31af14bcb3dc4a6499c90b18efb50ea6."
time="2023-03-17T09:35:14-07:00" level=info msg="Got VPC default SecurityGroupID r022-734525a9-0339-4272-8e99-932fcf3e0988."
time="2023-03-17T09:35:14-07:00" level=trace msg="Waiting for the VPC to be available before creating subnet..."
time="2023-03-17T09:35:14-07:00" level=trace msg="createSubnet()"
time="2023-03-17T09:35:15-07:00" level=info msg="Creating subnet sh-e2e-test-vpc-subnet in VPC r022-ef0d6365-b867-4c18-adaa-61fc17a71cd2 in Zone jp-tok-2."
time="2023-03-17T09:35:16-07:00" level=info msg="Created subnet with ID 02f7-4c0b513f-86f3-4fa8-8941-2d134dfeef4d."
time="2023-03-17T09:35:17-07:00" level=info msg="Public Gatwway r022-ce053028-47ee-45eb-a184-3f94735242f2 was created."
time="2023-03-17T09:35:18-07:00" level=info msg="Public Gatwway r022-ce053028-47ee-45eb-a184-3f94735242f2 was attached to Subnet 02f7-4c0b513f-86f3-4fa8-8941-2d134dfeef4d."
time="2023-03-17T09:35:18-07:00" level=trace msg="CreateCluster()"
time="2023-03-17T09:35:19-07:00" level=info msg="Creating cluster sh-e2e-test."
time="2023-03-17T09:35:29-07:00" level=info msg="Waiting for cluster sh-e2e-test to be available."
time="2023-03-17T09:35:29-07:00" level=info msg="Waited 0 minutes..."
time="2023-03-17T09:36:30-07:00" level=info msg="Waited 1 minutes..."
time="2023-03-17T09:37:31-07:00" level=info msg="Waited 2 minutes..."
time="2023-03-17T09:38:31-07:00" level=info msg="Waited 3 minutes..."
time="2023-03-17T09:39:32-07:00" level=info msg="Waited 4 minutes..."
time="2023-03-17T09:40:33-07:00" level=info msg="Waited 5 minutes..."
time="2023-03-17T09:41:34-07:00" level=info msg="Waited 6 minutes..."
time="2023-03-17T09:42:35-07:00" level=info msg="Waited 7 minutes..."
time="2023-03-17T09:43:35-07:00" level=info msg="Waited 8 minutes..."
time="2023-03-17T09:44:36-07:00" level=info msg="Waited 9 minutes..."
time="2023-03-17T09:45:37-07:00" level=info msg="Waited 10 minutes..."
time="2023-03-17T09:46:37-07:00" level=info msg="Waited 11 minutes..."
time="2023-03-17T09:47:38-07:00" level=info msg="Waited 12 minutes..."
time="2023-03-17T09:48:39-07:00" level=info msg="Waited 13 minutes..."
time="2023-03-17T09:49:40-07:00" level=info msg="Waited 14 minutes..."
time="2023-03-17T09:50:40-07:00" level=info msg="Waited 15 minutes..."
time="2023-03-17T09:51:41-07:00" level=info msg="Waited 16 minutes..."
time="2023-03-17T09:52:43-07:00" level=info msg="Cluster sh-e2e-test is available."
time="2023-03-17T09:52:43-07:00" level=info msg="Sync cluster kubeconfig to /root/.kube with current config context"
time="2023-03-17T09:52:46-07:00" level=info msg="Downloaded cluster sh-e2e-test kube configuration into folder /root/.kube"
Install CoCo and cloud-api-adaptor
Wait for the cc-operator-daemon-install DaemonSet be available
Wait for the pod cc-operator-daemon-install-9w84t be ready
Wait for the cloud-api-adaptor-daemonset DaemonSet be available
Wait for the pod cloud-api-adaptor-daemonset-k6mmh be ready
Wait for the kata runtimeclass be created
=== RUN   TestCreateSimplePod
=== RUN   TestCreateSimplePod/Simple_Peer_Pod
=== RUN   TestCreateSimplePod/Simple_Peer_Pod/PodVM_is_created
PodVM name:  simple-peer-pod
Instance number:  0  Instance id:  02f7_9f99423e-42ab-460c-9b29-c4108631a6d5  Instance name:  podvm-simple-peer-pod-4de276cc
--- PASS: TestCreateSimplePod (47.79s)
    --- PASS: TestCreateSimplePod/Simple_Peer_Pod (47.79s)
        --- PASS: TestCreateSimplePod/Simple_Peer_Pod/PodVM_is_created (1.92s)
PASS
ok  	github.com/confidential-containers/cloud-api-adaptor/test/e2e	1282.188s
```
Thanks Qifeng!

huoqifeng commented:
Thanks Steve and Yohei help debug it, I'll checking the log and try to figure out why some error happens in a new PR. I'll merge it firstly so that more people can try it and help find more potential issues.

yoheiueda commented:
I also got a successful result with an x86 VM image as follows. Thank you very much, @huoqifeng!

```
=== RUN   TestCreateSimplePod
=== RUN   TestCreateSimplePod/Simple_Peer_Pod
=== RUN   TestCreateSimplePod/Simple_Peer_Pod/PodVM_is_created
PodVM name:  simple-peer-pod
Instance number:  0  Instance id:  02e7_c55afd05-82f5-4891-9d94-b55c4579d177  Instance name:  podvm-simple-peer-pod-03238bdf
--- PASS: TestCreateSimplePod (60.26s)
    --- PASS: TestCreateSimplePod/Simple_Peer_Pod (60.26s)
        --- PASS: TestCreateSimplePod/Simple_Peer_Pod/PodVM_is_created (0.13s)
PASS
time="2023-03-18T07:46:17Z" level=info msg="Deleting Cluster e2e-test1."
time="2023-03-18T07:46:18Z" level=info msg="Waiting for cluster e2e-test1 to be removed..."
time="2023-03-18T07:46:18Z" level=info msg="Waited 0 minutes..."
time="2023-03-18T07:47:19Z" level=info msg="Waited 1 minutes..."
time="2023-03-18T07:48:19Z" level=info msg="Waited 2 minutes..."
time="2023-03-18T07:49:20Z" level=info msg="Waited 3 minutes..."
time="2023-03-18T07:50:20Z" level=info msg="Cluster e2e-test1 is removed."
time="2023-03-18T07:50:22Z" level=info msg="SSH Key yohei-ssh-e2e with ID r022-859b91b9-e3c5-450e-98c8-a5b0c58576c0 is deleted."
time="2023-03-18T07:50:23Z" level=info msg="Found subnet e2e-test1-vpc-subnet with ID 02e7-8e9561e1-5f40-4e13-bc64-c3b55aa3ac4b."
time="2023-03-18T07:50:24Z" level=info msg="Public Gatwway r022-c0a51059-e4f3-412c-a8d4-9d97f491839c was unattached from Subnet 02e7-8e9561e1-5f40-4e13-bc64-c3b55aa3ac4b."
time="2023-03-18T07:50:24Z" level=info msg="Public Gatwway r022-c0a51059-e4f3-412c-a8d4-9d97f491839c was deleted."
time="2023-03-18T07:50:24Z" level=info msg="Deleting subnet with ID 02e7-8e9561e1-5f40-4e13-bc64-c3b55aa3ac4b."
time="2023-03-18T07:50:26Z" level=info msg="Deleted subnet with ID 02e7-8e9561e1-5f40-4e13-bc64-c3b55aa3ac4b."
time="2023-03-18T07:50:26Z" level=info msg="Found VPC e2e-test1-vpc with ID r022-c261c647-8625-41a8-955e-1ed6c41f6a89."
time="2023-03-18T07:50:26Z" level=info msg="Deleting VPC with ID r022-c261c647-8625-41a8-955e-1ed6c41f6a89."
time="2023-03-18T07:50:26Z" level=info msg="Deleted VPC with ID r022-c261c647-8625-41a8-955e-1ed6c41f6a89."
ok  	github.com/confidential-containers/cloud-api-adaptor/test/e2e	440.710s
```
